Lib/imghdr.py

```diff
@@ -36,7 +36,7 @@ def what(file, h=None):
 
 def test_jpeg(h, f):
     """JPEG data in JFIF or Exif format"""
-    if h[6:10] in (b'JFIF', b'Exif'):
+    if h.startswith(b'\xff\xd8'):
         return 'jpeg'
```

Comment:
While technically this test is correct, there is a risk to have false positive. Would it be possible to design a stricter test?

My Hachoir parser for JPEG picture checks that it's able to parse at least 3 chunks:
https://github.com/vstinner/hachoir/blob/15f61aef34115a7ed1e76efc84b9176be55032a4/hachoir/parser/image/jpeg.py#L565-L578

A chunk must start with 0xFF, then have a tag in the 0xD0..0xD9 range.

Reply:
@vstinner  Hi,

I think I understand your code. For example in b'\xff\xd8\xff\xdb\x00C\....':

1. First 2 bytes must equal to '\xff\xd8'
2. Then you are looking for the tag (defined in JpegChunk.TAG_INFO) in the next 3 bytes( i.e. '\xff\xdb\x00C').

Am I correct?

Reply:
@vstinner Hi,

I had trouble understainding the code you provided. Can you please elebroate more?
Thanks.